test/unit/api-client-test.js

```diff
@@ -67,51 +252,254 @@ describe("ApiClient", function () {
+    it("should return all stories and collection id as cache tags for a manual collection", function () {
       const collection = Collection.build({
         id: "42",
-        "collection-cache-keys": ["sc/1/38586"],
+        "collection-cache-keys": ["c/1/42"],
         items: [
           {
             type: "story",
             story: { id: "264f46f9-e624-4718-b391-45e4bca7aeb6" },
           },
+          {
+            type: "story",
+            story: { id: "1e5acfe4-d8d2-40f8-a8f4-049cf1ac9fb4" },
+          },
         ],
       });
       assert.deepEqual(
-        ["c/1/42", "s/1/264f46f9", "sc/1/38586"],
+        ["c/1/42", "s/1/264f46f9", "s/1/1e5acfe4"],
         collection.cacheKeys(1)
       );
     });
 
-    it("adds nested collections when depth is passed", function () {
+    it("should include all nested stories and collection ids as cache tags for a manual collection", function () {
       const collection = Collection.build({
         id: "42",
-        "collection-cache-keys": ["sc/1/38586"],
+        type: "collection",
+        "collection-cache-keys": ["c/1/42"],
         items: [
           {
+            type: "story",
+            story: { id: "264f46f9-e624-4718-b391-45e4bca7aeb6" },
+          },
+          {
+            type: "story",
+            story: { id: "1e5acfe4-d8d2-40f8-a8f4-049cf1ac9fb4" },
+          },
+          {
+            id: "43",
             type: "collection",
+            "collection-cache-keys": ["c/1/43"],
+            items: [
+              {
+                type: "story",
+                story: { id: "0c2809d8-c60b-4cf2-b5eb-9aaeec90c062" },
+              },
+              {
+                type: "story",
+                story: { id: "fd4fa2c4-5441-434a-bf76-7660bbf1a09d" },
+              },
+            ],
+          }
+        ],
+      });
+      assert.deepEqual(
+        ["c/1/42", "s/1/264f46f9", "s/1/1e5acfe4", "c/1/43", "s/1/0c2809d8", "s/1/fd4fa2c4"],
+        collection.cacheKeys(1)
+      );
+    });
+
+    it("should return only collection cache keys excluding story keys as cache keys for an automated collection", function () {
+      const collection = Collection.build({
+        id: "42",
+        "collection-cache-keys": ["c/1/42", "sc/1/38586"],
+        automated: true,
+        items: [
+          {
+            type: "story",
+            story: { id: "264f46f9-e624-4718-b391-45e4bca7aeb6" },
+          },
+          {
+            type: "story",
+            story: { id: "1e5acfe4-d8d2-40f8-a8f4-049cf1ac9fb4" },
+          },
+        ],
+      });
+      assert.deepEqual(
+        ["c/1/42", "sc/1/38586"],
+        collection.cacheKeys(1)
+      );
+    });
+
+    it("should include all nested collection cache keys excluding story keys as cache keys for automated collection", function () {
+      const collection = Collection.build({
+        id: "42",
+        "collection-cache-keys": ["c/1/42", "sc/1/38586"],
```

Comment:
I know it it is too much of an ask at this point, but can this cache-key creation logic for a collection, automated or otherwise be done in the backend and FE just forwards it?

Also this same logic should apply to the API purges of the collections right? Do we have that logic in place for automated collections?

Reply:
Yes, that will work. Only thing is that will be a breaking change. The framework exposes this cachekeys as a function to which we can pass a depth. And if a client is passing a depth then the new change will not take that into account. If we have control over that then we can mitigate that as well.

Yes we do have. In the case of changing an automated collection (like adding a new collection, or ranking a story in collection) we are purging the collection alone. And when we publish a story, we are purging all the tags and sections that story belongs to. So this collection gets purged by CF.